markdown_it/main.py

```diff
@@ -89,21 +106,24 @@ def configure(self, presets: Union[str, Mapping]) -> "MarkdownIt":
         That will give better compatibility with next versions.
         """
         if isinstance(presets, str):
-            presetName = presets
-            presets = _PRESETS.get(presetName, None)
-            if not presets:
-                raise KeyError(
-                    'Wrong `markdown-it` preset "' + presetName + '", check name'
-                )
-        if not presets:
-            raise ValueError("Wrong `markdown-it` preset, can't be empty")
-        config = AttrDict(presets)
-
-        if "options" in config:
-            self.set(config.options)
+            config = _PRESETS.get(presets, None)
+            if not config:
+                raise KeyError(f"Wrong `markdown-it` preset '{presets}', check name")
+        else:
+            config = presets
+
+        if not config:
+            raise ValueError("Wrong `markdown-it` config, can't be empty")
+
+        options = config.get("options", {}) or {}
+        if options_update:
+            options = {**options, **options_update}
```

Comment:
I'm fairly sure the JS upstream does a deep/recursive merge here. Might want to port that code?

E.g merging `{"a": {"b": 0, "c": 1}}` with `{"a": {"b": 99}}` will result in `{"a": {"b": 99}}` but I think `{"a": {"b": 99, "c": 1}}` is expected

Reply:
Ah sorry it seems I misremember, upstream only does a "1st level merge" just like we do here.